services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/api_client/_thin.py

```diff
@@ -0,0 +1,224 @@
+    ) -> Response:
+        url = self._get_url(dynamic_sidecar_endpoint, "/containers/directory-watcher")
+        return await self._client.patch(url, json=dict(is_enabled=is_enabled))
+
+    @retry_on_errors
+    @expect_status(status.HTTP_204_NO_CONTENT)
+    async def post_containers_ports_outputs_dirs(
+        self, dynamic_sidecar_endpoint: AnyHttpUrl, *, outputs_labels: dict[str, Any]
+    ) -> Response:
+        url = self._get_url(dynamic_sidecar_endpoint, "/containers/ports/outputs/dirs")
+        return await self._client.post(url, json=dict(outputs_labels=outputs_labels))
+
+    @retry_on_errors
+    @expect_status(status.HTTP_200_OK)
+    async def post_containers_ports_outputs_pull(
+        self,
+        dynamic_sidecar_endpoint: AnyHttpUrl,
+        *,
+        port_keys: Optional[list[str]] = None,
+    ) -> Response:
+        url = self._get_url(dynamic_sidecar_endpoint, "/containers/ports/outputs:pull")
+        return await self._client.post(
+            url, json=port_keys, timeout=self._save_restore_timeout
+        )
+
+    @retry_on_errors
+    @expect_status(status.HTTP_204_NO_CONTENT)
+    async def post_containers_ports_outputs_push(
+        self,
+        dynamic_sidecar_endpoint: AnyHttpUrl,
+        *,
+        port_keys: Optional[list[str]] = None,
+    ) -> Response:
+        url = self._get_url(dynamic_sidecar_endpoint, "/containers/ports/outputs:push")
+        return await self._client.post(
+            url, json=port_keys, timeout=self._save_restore_timeout
+        )
+
+    @retry_on_errors
+    @expect_status(status.HTTP_200_OK)
+    async def get_containers_name(
+        self, dynamic_sidecar_endpoint: AnyHttpUrl, *, dynamic_sidecar_network_name: str
+    ) -> Response:
+        filters = json.dumps({"network": dynamic_sidecar_network_name})
+        url = self._get_url(
+            dynamic_sidecar_endpoint, f"/containers/name?filters={filters}"
+        )
+        return await self._client.get(url=url)
+
+    @retry_on_errors
+    @expect_status(status.HTTP_204_NO_CONTENT)
+    async def post_containers_restart(
+        self, dynamic_sidecar_endpoint: AnyHttpUrl
+    ) -> Response:
+        url = self._get_url(dynamic_sidecar_endpoint, "/containers:restart")
+        return await self._client.post(url, timeout=self._restart_containers_timeout)
+
+    @retry_on_errors
+    @expect_status(status.HTTP_204_NO_CONTENT)
+    async def post_containers_networks_attach(
+        self,
+        dynamic_sidecar_endpoint: AnyHttpUrl,
+        *,
+        container_id: str,
+        network_id: str,
+        network_aliases: list[str],
+    ) -> Response:
+        url = self._get_url(
+            dynamic_sidecar_endpoint, f"/containers/{container_id}/networks:attach"
+        )
+        return await self._client.post(
+            url,
+            json=dict(network_id=network_id, network_aliases=network_aliases),
+            timeout=self._attach_detach_network_timeout,
+        )
+
+    @retry_on_errors
```

Comment:
I have a question regarding all these retries. 
Now that I see that the dynamic scheduler actually already retries forever... what does this retrial mechanism in the client bring in here? and is that not maybe creating long calls in case there is a real issue,could this trigger the dynamic scheduler in triggering too many tasks and then being locked at doing so?

Reply:
The dynamic scheduler can only have **one task** per service **at a time running** which interacts with the dynamic-sidecar. This way, even if you try to schedule a new task every 5 seconds, if there already is one in place interacting, none will be started.

Long running operations are not an issue. Retrying the communication will not create further issues.

Reply:
👍